runners/direct-java/src/main/java/org/apache/beam/runners/direct/TransformExecutor.java

```diff
@@ -138,13 +141,6 @@ private void processElements(
 
         evaluator.processElement(value);
 
-        // Report the physical metrics after each element
-        MetricUpdates deltas = metricsContainer.getUpdates();
-        if (deltas != null) {
-          context.getMetrics().updatePhysical(inputBundle, deltas);
```

Comment:
Sounds good -- created https://issues.apache.org/jira/browse/BEAM-2186 to track direct runner updates.